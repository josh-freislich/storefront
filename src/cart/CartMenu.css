div.cart-menu {
  position: fixed;
  right: 10px;
  width: 370px;
  padding: 0 15px;
  overflow: hidden;
  background-color: #FFF;
  text-align: left;
  border-color: #DDDDDD;
  border-style: solid;
  border-width: 0 1px 1px;
  max-height: 0;

  -webkit-box-shadow: 0px 10px 10px 0px rgba(0,0,0,0.08);
  -moz-box-shadow: 0px 10px 10px 0px rgba(0,0,0,0.08);
  box-shadow: 0px 10px 10px 0px rgba(0,0,0,0.08);

  -webkit-transition:max-height 500ms ease;
  -moz-transition:max-height 500ms ease;
  -o-transition:max-height 500ms ease;
  transition:max-height 500ms ease;
}

div.cart-menu.open {
  max-height: 100%;
}

div.cart-menu a {
  padding: 0 !important;
}

div.empty-menu-list {
  display: none;
}
.cart-empty div.empty-menu-list {
  display: block;
  height: 420px;
  background-image: url(/media/empty-cart.png);
  background-repeat: no-repeat;
  background-position: center 50px;
  text-align: center;
}

.cart-empty div.empty-menu-list div.empty {
  padding: 200px 0 10px;
  text-align: center;
  color: #777777;
}

.cart-empty div.empty-menu-list button.secondary {
  margin-top: 10px;
  width: 220px;
  border: 1px solid #333;
  height: 55px;
  font-size: 16px;
}

div.cart-menu .menu-list {
    height: 300px;
    overflow: auto;
}

div.cart-menu.cart-empty .menu-list,
div.cart-menu.cart-empty .cart-menu-total,
div.cart-menu.cart-empty .cart-menu-buttons  {display: none;}


div.cart-menu .product-item {
  grid-template-columns: 90px auto 30px;
  padding: 15px 0;
  position: relative;
}

div.cart-menu .product-item > div {
  text-align: left;
  padding: 0 10px;
  display: grid;
}

div.cart-menu .product-item h1 {
  font-family: arial, sans-serif;
  font-size: 14px;
  text-transform: uppercase;
  padding: 0;
}

div.cart-menu .product-item .name {
  grid-row: 1;
  margin: 0;
  text-align: left;
  line-height: 20px;
}
div.cart-menu .product-item .name .count {
  display: inline-block;
}

div.cart-menu .product-item .brand {
  grid-row: 2;
}

div.cart-menu .product-item .brand {
  font-family: arial, sans-serif;
  color: #999999;
  font-size: 10px;
  padding: 0 0 5px;
}

div.cart-menu .product-item .price {
  font-family: arial, sans-serif;
  color: #444444;
  font-size: 12px;
  font-weight: 600;
}

div.cart-menu .product-item .color,
div.cart-menu .product-item .description {
  display: none;
}

div.cart-menu .product-item .remove-item {
  grid-column: 3;
  grid-row: 1;
  border: 0 none;
  color: #999999;
  font-size: 16px;
  padding: 0;
  margin: 0;
  outline: none;
  cursor: pointer;
  position: absolute;
  top: -3px;
  right: 15px;
}






.cart-menu .cart-menu-total {
  margin: 0 0 20px;
  border-top: 1px solid #DDD;
  padding-top: 18px;
  color: #CCCCCC;
}
.cart-menu .cart-menu-total .cart-menu-total-amount {
  position: absolute;
  right: 15px;
}

.cart-menu .cart-menu-buttons {
  display: grid;
  grid-template-columns: 47% 47%;
  grid-column-gap: 6%;
}
.cart-menu .cart-menu-buttons a {
  line-height: initial;
}
.cart-menu button.view-cart,
.cart-menu button.checkout {
  border: 0 none;
  width: 100%;
  height: 55px;
  cursor: pointer;
  font-weight: 600;
}

.cart-menu button.view-cart {
  margin: 0 15px 15px 0;
  border: 1px solid #333333;
  color: #333333;
}

.cart-menu button.checkout {
  background-color: #333333;
  color: #FFFFFF;
}
