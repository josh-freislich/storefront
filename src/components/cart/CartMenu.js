import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {sumMoney, formatMoney} from '../../utils'
import ProductItem from '../product/ProductItem'

class CartMenu extends Component {
  render() {
    const {cart = [], open, empty, remove} = this.props;

    const cartStyles = [
      'cart-menu',
      empty ? 'cart-empty' : '',
      open ? 'open' : ''
    ].join(' ');

    const productList = cart.map(item => item.count ? (
      <ProductItem
        key={item.product.title}
        product={item.product}
        count={item.count}
        rightChildren={<button
          onClick={ () => remove(item.product) }
          className="remove-item">x</button>}
        />
    ) : null);

    const prices = cart.map(i => i.product.price * i.count);
    const total = sumMoney(...prices);

    return (
      <div className={cartStyles}>
        <div className="empty-menu-list">
          <div className="empty">Your cart is empty</div>
          <div className="ideas">Looking for ideas?</div>
          <Link to="/"><button className="secondary">See trending items</button></Link>
        </div>

        <div className="menu-list">
          {productList}
        </div>

        <div className="cart-menu-total">
          <span className="cart-menu-total-description">TOTAL</span>
          <span className="cart-menu-total-amount">${formatMoney(total)}</span>
        </div>

        <div className="cart-menu-buttons">
          <Link to="/cart"><button className="view-cart">VIEW CART</button></Link>
          <button className="checkout">CHECKOUT</button>
        </div>
      </div>
    );
  }
}

export default CartMenu;
