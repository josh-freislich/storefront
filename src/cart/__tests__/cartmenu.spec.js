import React from 'react';
import CartMenu from '../CartMenu';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const products = {
  kiriko: {
    "title": "Blue Stripe Stoneware Plate",
    "brand": "Kiriko",
    "price": 40,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "blue-stripe-stoneware-plate.jpg",
    "color": ""
  },
  heme :   {
    "title": "Heme",
    "brand": "Dust & Form",
    "price": 52,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    "image": "heme.jpg",
    "color": ""
  }
}

it('<CartMenu /> renders correctly when default closed state', () => {
  const wrapper = shallow(
    <CartMenu />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('<CartMenu /> renders correctly when open state', () => {
  const wrapper = shallow(
    <CartMenu
      open={true}
      empty={true}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('<CartMenu /> renders correctly when has products', () => {
  const cart = [ { count: 2, product: products.kiriko }, { count: 1, product: products.heme } ]
  const wrapper = shallow(
    <CartMenu
      open={true}
      cart={cart}
    />

  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
