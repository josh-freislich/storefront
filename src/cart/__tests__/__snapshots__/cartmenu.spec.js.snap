// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartMenu /> renders correctly when default closed state 1`] = `
<div
  className="cart-menu  "
>
  <div
    className="empty-menu-list"
  >
    <div
      className="empty"
    >
      Your cart is empty
    </div>
    <div
      className="ideas"
    >
      Looking for ideas?
    </div>
    <Link
      to="/"
    >
      <button
        className="secondary"
      >
        See trending items
      </button>
    </Link>
  </div>
  <div
    className="menu-list"
  />
  <div
    className="cart-menu-total"
  >
    <span
      className="cart-menu-total-description"
    >
      TOTAL
    </span>
    <span
      className="cart-menu-total-amount"
    >
      $
      0.00
    </span>
  </div>
  <div
    className="cart-menu-buttons"
  >
    <Link
      to="/cart"
    >
      <button
        className="view-cart"
      >
        VIEW CART
      </button>
    </Link>
    <button
      className="checkout"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;

exports[`<CartMenu /> renders correctly when has products 1`] = `
<div
  className="cart-menu  open"
>
  <div
    className="empty-menu-list"
  >
    <div
      className="empty"
    >
      Your cart is empty
    </div>
    <div
      className="ideas"
    >
      Looking for ideas?
    </div>
    <Link
      to="/"
    >
      <button
        className="secondary"
      >
        See trending items
      </button>
    </Link>
  </div>
  <div
    className="menu-list"
  >
    <ProductItem
      count={2}
      key="Blue Stripe Stoneware Plate"
      product={
        Object {
          "brand": "Kiriko",
          "color": "",
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
          "image": "blue-stripe-stoneware-plate.jpg",
          "price": 40,
          "title": "Blue Stripe Stoneware Plate",
        }
      }
      rightChildren={
        <button
          className="remove-item"
          onClick={[Function]}
        >
          x
        </button>
      }
    />
    <ProductItem
      count={1}
      key="Heme"
      product={
        Object {
          "brand": "Dust & Form",
          "color": "",
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
          "image": "heme.jpg",
          "price": 52,
          "title": "Heme",
        }
      }
      rightChildren={
        <button
          className="remove-item"
          onClick={[Function]}
        >
          x
        </button>
      }
    />
  </div>
  <div
    className="cart-menu-total"
  >
    <span
      className="cart-menu-total-description"
    >
      TOTAL
    </span>
    <span
      className="cart-menu-total-amount"
    >
      $
      132.00
    </span>
  </div>
  <div
    className="cart-menu-buttons"
  >
    <Link
      to="/cart"
    >
      <button
        className="view-cart"
      >
        VIEW CART
      </button>
    </Link>
    <button
      className="checkout"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;

exports[`<CartMenu /> renders correctly when open state 1`] = `
<div
  className="cart-menu cart-empty open"
>
  <div
    className="empty-menu-list"
  >
    <div
      className="empty"
    >
      Your cart is empty
    </div>
    <div
      className="ideas"
    >
      Looking for ideas?
    </div>
    <Link
      to="/"
    >
      <button
        className="secondary"
      >
        See trending items
      </button>
    </Link>
  </div>
  <div
    className="menu-list"
  />
  <div
    className="cart-menu-total"
  >
    <span
      className="cart-menu-total-description"
    >
      TOTAL
    </span>
    <span
      className="cart-menu-total-amount"
    >
      $
      0.00
    </span>
  </div>
  <div
    className="cart-menu-buttons"
  >
    <Link
      to="/cart"
    >
      <button
        className="view-cart"
      >
        VIEW CART
      </button>
    </Link>
    <button
      className="checkout"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
